src/app/context_observer.h

```diff
@@ -15,8 +15,9 @@ namespace app {
 
   class ContextObserver {
   public:
-    virtual ~ContextObserver() { }
-    virtual void onActiveSiteChange(const Site& site) { }
+    virtual ~ContextObserver() = default;
+    virtual void onActiveSiteChange(const Site& site) { };
+    virtual void onBeforeActiveSiteChange(const Site& fromSite) { };
```

Comment:
As a side note, it's not necessary to change the destructor to `= default` and even more, the `{ }` is shorter :upside_down_face: Also `;` is not needed in these cases:
```suggestion
    virtual ~ContextObserver() { }
    virtual void onActiveSiteChange(const Site& site) { }
    virtual void onBeforeActiveSiteChange(const Site& fromSite) { }
```